packages/headless/src/features/search/search-actions.ts

```diff
@@ -104,13 +104,24 @@ export const updateSearchAction = createAction<SearchAction | undefined>(
   'search/updateSearchAction'
 );
 
+export const setReadyToGenerateAnswerAction = createAction<boolean>(
+  'search/setReadyToGenerateAnswerAction'
+);
+
 export const executeSearch = createAsyncThunk<
   ExecuteSearchThunkReturn,
   TransitiveSearchAction,
   AsyncThunkSearchOptions<StateNeededByExecuteSearch>
 >(
   'search/executeSearch',
   async (searchAction: TransitiveSearchAction, config) => {
+    // Set the search action cause in the state before executing the request.
+    // This makes it available for subscribers (e.g., Generated Answer controller)
+    // After setting the cause, mark the state as ready to generate an answer.
+    // This guarantees that answer generation only happens once the cause is set.
+    config.dispatch(updateSearchAction(searchAction.next));
+    config.dispatch(setReadyToGenerateAnswerAction(true));
```

Comment:
Simply changing the `searchAction` before triggering the execute search itself should be enough so that listeners are ready to get the `searchAction` when listening to when a new search gets triggered.

Otherwise we need a different solution.